bootstrap/bootstrap.sh

```diff
@@ -28,8 +29,54 @@ sudo apt-get update
 sudo apt-get -y dist-upgrade
 
 #----nodejs install
-curl -sL https://deb.nodesource.com/setup_7.x | sudo -E bash -
+curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
 sudo apt-get install -y nodejs
 
-#----alsa install
-sudo apt-get install -y alsa-base alsa-utils
+#----install official requirements
+sudo apt-get install -y alsa-base alsa-utils libasound2-dev
+
+#----install missing pigpio in Raspbian Lite (the command npm install pigpio will be exec by package.json)
+sudo apt-get install pigpio
+
+#----install git and download tjbot
+if [ ! -d "${TJBOT_FOLDER}/recipes/conversation" ]; then
+    sudo apt-get install -y git
+
+    rm -Rf /home/pi/git/tjbot
+    git clone https://github.com/ibmtjbot/tjbot.git /home/pi/git/tjbot
+
+    TJBOT_FOLDER='/home/pi/git/tjbot'
+fi
+
+#----install conversation (to install it will resolve tjbotlib and other dependencies)
+cd $TJBOT_FOLDER
+cd recipes/conversation
+echo "path: $PWD"
+
+sudo npm install --unsafe-perm -g node-gyp
```

Comment:
> permission errors (even using sudo) that could be fixed using --unsafe-perm
> I tried to know what is happening with sudo npm install or npm install but I didn't know. 

It's not easy to understand what npm does when called with sudo when you are installing native modules that need to be compiled. I'll attempt to explain it.

If you cd to the directory of a native module that needs to be compiled and call the command `sudo npm install` in that directory npm will NOT execute everything with root privileges. In particular, when running the script to compile the native module, the process that does the compile has the privileges of user `nobody`. This is not a lot of privileges and explains why errors such as the following are displayed:

```
gyp WARN EACCES user "root" does not have permission to access the dev dir "/root/.node-gyp/6.10.2"
```

npm does this as running scripts is security risk. The workaround is to run the command `sudo npm install --unsafe-perm` instead. This gives the process that does the compile root privileges.

The npm documentation mentions the topic [here](https://docs.npmjs.com/misc/config#unsafe-perm), [here](https://docs.npmjs.com/misc/config#user) and [here](https://docs.npmjs.com/misc/scripts#user).

That being said, I don't see any reason why `sudo npm install --unsafe-perm` needs to be called here. `npm install` should be enough. If it's not enough something is messed up on your system.

> and also a lot of warnings and deprecated messages
> But that a lot of warning / deprecated messages (https://gist.github.com/fivdi/f7a45e595d3039cb0c5d631425bb080d) give the impression that are something wrong. A end user could thing that it is broken.

Yes, this is correct. Several native modules are being installed and lots of them result in warnings being displayed. But they are only warnings, not errors. I hope you noticed that `pigpio` compiles cleanly and doesn't display any warnings :). I don't think you'll be able to do much about this other than document that fact that the warnings are displayed so that users feel more at ease.

> So sudo npm install --unsafe-perm -g readline-sync run tests fine without errors

I don't understand this. Haw can running the command `sudo npm install --unsafe-perm -g readline-sync` run tests?

> and also sudo npm install --unsafe-perm -g node-gyp do the same with conversation.

I don't understand this either. How can running the command `sudo npm install --unsafe-perm -g node-gyp` run tests?